### PR TITLE
[google_maps_flutter] Fix UIKit availability and pod lint warnings

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Fix google_maps_flutter UIKit availability warnings introduced in https://github.com/flutter/plugins/pull/2607.  Also fix pod lib lint warnings:
```
 -> google_maps_flutter (0.0.1)
    - WARN  | license: Missing license type.
    - WARN  | description: The description is equal to the summary.
    - WARN  | [iOS] keys: Missing primary key for `source` attribute. The acceptable ones are: `git, hg, http, svn`.
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:176:7: warning: 'UIGraphicsImageRendererFormat' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:176:78: warning: 'defaultFormat' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:176:48: warning: 'UIGraphicsImageRendererFormat' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:178:7: warning: 'UIGraphicsImageRenderer' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:179:13: warning: 'UIGraphicsImageRenderer' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:181:53: warning: 'UIGraphicsImageRendererContext' is only available on iOS 10.0 or newer [-Wunguarded-availability]
...
[!] google_maps_flutter did not pass validation, due to 9 warnings (but you can use `--allow-warnings` to ignore them).
```

## Related Issues
Fixes https://github.com/flutter/flutter/issues/55258
https://github.com/flutter/flutter/issues/55245
Dependency to merge https://github.com/flutter/plugin_tools/pull/97

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m [hunk 1]
```diff
@@ -172,20 +172,24 @@ - (void)onMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
...
-      UIGraphicsImageRendererFormat* format = [UIGraphicsImageRendererFormat defaultFormat];
-      format.scale = [[UIScreen mainScreen] scale];
-      UIGraphicsImageRenderer* renderer =
-          [[UIGraphicsImageRenderer alloc] initWithSize:_mapView.frame.size format:format];
-
-      UIImage* image = [renderer imageWithActions:^(UIGraphicsImageRendererContext* context) {
-        [_mapView.layer renderInContext:context.CGContext];
-      }];
-      result([FlutterStandardTypedData typedDataWithBytes:UIImagePNGRepresentation(image)]);
+    if (@available(iOS 10.0, *)) {
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:175]: Matched the pattern at https://github.com/flutter/plugins/blob/5ca81a351d34558380e2f9f9ec9f6b6205435a49/packages/webview_flutter/ios/Classes/FLTCookieManager.m#L45.

Comment (cyanglaz) [reply packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:175]: Thanks for catching this!

File: packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m [hunk 2]
```diff
@@ -172,20 +172,24 @@ - (void)onMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
...
+      } else {
+        result([FlutterError errorWithCode:@"GoogleMap uninitialized"
+                                   message:@"takeSnapshot called prior to map initialization"
+                                   details:nil]);
+      }
     } else {
-      result([FlutterError errorWithCode:@"GoogleMap uninitialized"
-                                 message:@"takeSnapshot called prior to map initialization"
-                                 details:nil]);
+      NSLog(@"Taking snapshots is not supported for Flutter Google Maps prior to iOS 10.");
```
Comment (jmagman) [inline packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:192]: I think I need to call `result` here.

Comment (cyanglaz) [reply packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m:192]: oh yeah.  Missed it. Thanks!

**Comment by andrewescutia** [comment]
since this isnt a breaking change would it be worthy of cherry picking it to a more stable branch? Not sure how that process works exactly.

**Comment by jmagman** [comment]
@andrewescutia This change is already available in google_maps_flutter-v0.5.26+2 or greater, if you update your app's pubspec version range.

**Comment by andrewescutia** [comment]
noice! I see that now within the pub changelog. For some reason I was reading this as an update to Flutter itself. Do the PRs themselves notate what package version they are a part of?

**Comment by jmagman** [comment]
> noice! I see that now within the pub changelog. For some reason I was reading this as an update to Flutter itself. Do the PRs themselves notate what package version they are a part of?

Yes, each PR should update the CHANGELOG and version: https://github.com/flutter/plugins/pull/2681/files#diff-40608137945b7a665e28d032c6e04916R1

**Pull request merged**